internal/pkg/v2/infrastructure/redis/reading.go

```diff
@@ -0,0 +1,63 @@
+//
```

Comment:
@andresrinivasan needs to review this code.

Reply:
Apologies @lenny-intel. Will get to this later this week.

Reply:
Ditto on the curiosity point from `redis.event`.

The duplication of code in the two case sections caught my attention. I suggest a refactor.

The error for the `default:` case is currently using `ErrUnsupportedDatabase`. I don't believe this error is an unknown database type error.

Reply:
As model.Reading needs to be converted to two concrete Reading structs `model.BinaryReading` and `model.SimpleReading` and then the value can be retrieved from the structs, it's a bit difficult here to not avoid duplicate code. Would you please suggest if any work around?

As for the `default` case, a new error type `db.ErrUnsupportedReading` is added for the case. Thanks a lot for catching this issue.

Reply:
I suggest refactoring the common code in to a function.

Reply:
I agree, majority of the code is identical in each case, just create the `newReading` in each case and then move the common code to after the switch, before the return.

Reply:
ok, we are refactoring it